File: clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn grade [school grade]
  (school grade []))

(defn add [school name grade]
  (assoc school grade (conj (school grade []) name)))

(defn sorted [school]
  (into (sorted-map) (for [[score names] school] [score (sort names)])))

File: python/grade-school/grade_school.py
"""Create a roster for a School"""

# pylint: disable=missing-function-docstring

class School:  # pylint: disable=missing-class-docstring
    def __init__(self):
        self._grades = {}
        self.add_list = []

    def add_student(self, name, grade):
        if name not in self.roster():
            self._grades[grade] = self._grades.get(grade, []) + [name]
            self.add_list.append(True)
        else:
            self.add_list.append(False)  

    def roster(self):
        return [
            name
            for grade in sorted(self._grades)
            for name in sorted(self._grades[grade])
        ]

    def grade(self, grade_number):
        return sorted(self._grades.get(grade_number, []))

    def added(self):
        return self.add_list
